### PR TITLE
[Tokenomics] feat: add `dao_reward_address` param to tokenomics module

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

Add `dao_reward_address` param to the tokenomics module.

## Issue

- `TODO_BETA`

## Type of change

Select one or more from the following:

- [ ] New feature, functionality or library
- [x] Consensus breaking; add the `consensus-breaking` label if so. See #791 for details
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Documentation**: `make docusaurus_start`; only needed if you make doc changes
- [x] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- [x] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [x] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: config.yml [hunk 1]
```diff
@@ -284,6 +284,7 @@ genesis:
         mint_allocation_supplier: 0.7
         mint_allocation_source_owner: 0.15
         mint_allocation_application: 0.0
+        dao_reward_address: "pokt1eeeksh2tvkh7wzmfrljnhw4wrhs55lcuvmekkw"
```
Comment (bryanchriswhite) [inline config.yml:287]:
```suggestion
        # The dao reward address SHOULD match that of the "pnf" below (i.e. `make poktrolld_addr ACC_NAME=pnf`).
        # This is the address that will receive the dao/foundation rewards during claim settlement (global mint TLM).
        dao_reward_address: "pokt1eeeksh2tvkh7wzmfrljnhw4wrhs55lcuvmekkw"
```

File: config.yml [hunk 2]
```diff
@@ -286,6 +286,9 @@ genesis:
         mint_allocation_supplier: 0.7
         mint_allocation_source_owner: 0.15
         mint_allocation_application: 0.0
+        # The dao reward address SHOULD match that of the "pnf" below (i.e. `make poktrolld_addr ACC_NAME=pnf`).
```
Comment (Olshansk) [inline config.yml:289]: Can you add a `TODO_MAINNET(@Olshansk): Consolidate the usage of DAO/PNF throughout the configs & codebase`

File: x/tokenomics/types/params.go [hunk 3]
```diff
@@ -201,3 +223,32 @@ func ValidateMintAllocationSum(params *Params) error {
...
 }
+
+// ValidateDaoRewardAddress validates the DaoRewardAddress param.
+func ValidateDaoRewardAddress(daoRewardAddress any) error {
+	daoRewardAddressStr, ok := daoRewardAddress.(string)
+	if !ok {
+		return ErrTokenomicsParamInvalid.Wrapf("invalid parameter type: %T", daoRewardAddress)
+	}
+
+	// DEV_NOTE: Empty string is valid if mint_allocation_dao is 0.
```
Comment (Olshansk) [inline x/tokenomics/types/params.go:234]: Hmmm, why?

Any chance we can just prevent this from ever being zero altogether and avoid bugs or special/edge cases?

File: x/tokenomics/types/params.go [hunk 4]
```diff
@@ -201,3 +223,32 @@ func ValidateMintAllocationSum(params *Params) error {
...
+
+	if _, err := sdk.AccAddressFromBech32(daoRewardAddressStr); err != nil {
+		return ErrTokenomicsParamInvalid.Wrapf("invalid dao reward address %q: %s", daoRewardAddressStr, err)
+	}
+
+	return nil
+}
+
+// ValidateDaoRewardParams validates that the dao_reward_address param is not an
+// empty string so long as the mint_allocation_dao param is not 0.
```
Comment (Olshansk) [inline x/tokenomics/types/params.go:247]: `// empty string so long as the mint_allocation_dao param is not 0.`

I understand WHY you did this, but I'd rather keep it simpler. Less prone to bugs.

File: x/tokenomics/types/message_update_param.go [hunk 5]
```diff
@@ -66,16 +68,33 @@ func (msg *MsgUpdateParam) ValidateBasic() error {
...
 func (msg *MsgUpdateParam) paramTypeIsFloat() error {
-	if _, ok := msg.AsType.(*MsgUpdateParam_AsFloat); !ok {
+	return genericParamTypeIs[*MsgUpdateParam_AsFloat](msg)
+}
+
+func (msg *MsgUpdateParam) paramTypeIsString() error {
+	return genericParamTypeIs[*MsgUpdateParam_AsString](msg)
+}
+
+// TODO_TECHDEBT:
```
Comment (Olshansk) [inline x/tokenomics/types/message_update_param.go:89]:
```suggestion
// TODO_TECHDEBT(@bryanchriswhite):
```

File: x/tokenomics/types/message_update_param.go [hunk 6]
```diff
@@ -66,16 +68,33 @@ func (msg *MsgUpdateParam) ValidateBasic() error {
...
+// 1. Move this to a shared package.
+// 2. Refactor other module mesagese_upddate_param.go files to use this.
```
Comment (Olshansk) [inline x/tokenomics/types/message_update_param.go:91]:
```suggestion
// 2. Refactor other module message_update_param.go files to use this.
```

**Review by Olshansk — CHANGES_REQUESTED** [review]

**Pull request merged**